db/db_utils.go

```diff
@@ -413,19 +413,16 @@ func AggregateWithHint(collection string, pipeline interface{}, hint interface{}
 	return errors.WithStack(pipe.All(out))
 }
 
-// AggregateWithMaxTime runs aggregate and specifies a max query time which ensures the query won't go on indefinitely when the request is cancelled.
-// A db.Aggregation is returned because in some usages the same db.Aggregation needs to be reused after the pipeline results have been retrieved.
-func AggregateWithMaxTime(collection string, pipeline interface{}, out interface{}, maxTime time.Duration) (db.Aggregation, error) {
+// AggregateWithMaxTime runs aggregate and specifies a max query time which
+// ensures the query won't go on indefinitely when the request is cancelled.
+func AggregateWithMaxTime(collection string, pipeline interface{}, out interface{}, maxTime time.Duration) error {
```

Comment:
I stopped returning `pipe` because it is an iterator and cannot be reused after it exhausted.